src/components/boost-slider.js

```diff
@@ -32,11 +38,42 @@ class BoostSlider extends Component {
     this.setState({
       selectedBoostAmountUsd
     })
+
     $('[data-toggle="tooltip"]').tooltip()
+
+    this.checkOgnTransferAllowance()
+    this.onChange(this.state.selectedBoostAmount)
   }
 
   componentWillUnmount() {
     $('[data-toggle="tooltip"]').tooltip('dispose')
+    clearInterval(this.ognBalancePoll)
+  }
+
+  async checkOgnTransferAllowance() {
+    const userAddress = await origin.contractService.currentAccount()
+    const allowanceRemaining = await origin.token.getAllowance(userAddress)
+    console.log('Allowance Remaining: ', allowanceRemaining)
+    this.setState({
+      allowanceRemaining
+    })
+  }
+
+  async approveOgnTransfer() {
+    const ognAmount = this.state.selectedBoostAmount
+    const weiAmount = (ognAmount * 10 ** 18).toString()
```

Comment:
Let's us BigNumber or [web3.utils.toWei](https://web3js.readthedocs.io/en/1.0/web3-utils.html#towei).
Also nit: from a crypto linguistic standpoint it is not elegant to call this **wei** since we are not dealing with ETH here but with OGN tokens... I think the crypo party recommends to call this "[natural unit](https://theethereum.wiki/w/index.php/Natural_Units)"... :)

Reply:
@franckc @jordajm there was some discussion in April about whether or not DApp developers should have to deal with BigNumbers and these calculations. As I recall, we intentionally abstracted all of this wei conversion to origin-js and changed all of the return values to normal integers. [I documented this at the time](https://github.com/OriginProtocol/origin-docs/commit/897c84ba35727177b3e4555bc899ee2d029c5322) but I'm not sure where the discussion happened.

Reply:
I understand that we don't want to use floating point, but is there any other way that the DApp can avoid having to know about wei calculations?

Reply:
I'll take a look at what it would take to move this conversion into origin-js

Reply:
Thanks @jordajm I didn't dig very deep, but I don't see any uses of BigNumber or toWei in origin-dapp right now. So hopefully we can handle OGN similarly to however we're handling ETH right now. 🙏

Reply:
Ah, there is a `fromWei` in the ETH wallet action and some math in the OGN wallet action. cc @cuongdo

Reply:
Yeah. the OGN math should probably be in the Token resource.